beacon_node/eth1/src/block_cache.rs

```diff
@@ -36,11 +37,30 @@ impl Eth1Block {
     }
 }
 
+#[derive(Encode, Decode, Clone)]
+pub struct SszBlockCache {
+    block_cache: Vec<Eth1Block>,
+}
+
+impl SszBlockCache {
+    pub fn from_block_cache(cache: &BlockCache) -> Self {
+        Self {
+            block_cache: cache.blocks.clone(),
+        }
+    }
+
+    pub fn to_block_cache(&self) -> Result<BlockCache, String> {
+        Ok(BlockCache {
+            blocks: self.block_cache.clone(),
+        })
+    }
+}
+
 /// Stores block and deposit contract information and provides queries based upon the block
 /// timestamp.
 #[derive(Debug, PartialEq, Clone, Default)]
 pub struct BlockCache {
-    blocks: Vec<Eth1Block>,
+    pub blocks: Vec<Eth1Block>,
```

Comment:
Can we get away with not making this `pub`? This helps us make stronger guarantees about the consistency of the `BlockCache` (i.e., it can only be modified used the available `pub` functions).

Reply:
Yup sorry! Keep forgetting that non public fields can still be accessed within the same module :sweat_smile: